### DISCUSSION TITLE
When to use the static version of a method vs the non-static version

### DISCUSSION THREAD

**Discussion by marcusx2** [firebase/firebase-js-sdk#7625] [Q&A] [answered]
In this [closed issue](https://github.com/firebase/firebase-js-sdk/issues/6311), @hsubox76 [mentions](https://github.com/firebase/firebase-js-sdk/issues/6311#issuecomment-1142394436) that the static version of the method for authstatechanged and beforestatechanged should be preferred over the non-static version. Does this hold true for every non-static version of a method where a static version exists? For example, the [static reload](https://firebase.google.com/docs/reference/js/auth.md#reload) vs [non-static reload](https://firebase.google.com/docs/reference/js/auth.user.md#userreload). Why is the static version preferred over the non-static version, is it not just a matter of preference?

**Comment by ryanmio** [comment, marked as answer]
Use static methods for utility, non-static for object state.

### Static Methods
Use when operation doesn't depend on object's state.
Often more performant; no need to instantiate object.

### Non-Static Methods
Use when operation depends on object's state.
Allows method chaining on the instance.

In context of authstatechanged and beforestatechanged, static methods likely recommended for state-independent utility.
